apps/ewallet_db/lib/ewallet_db/export.ex

```diff
@@ -0,0 +1,191 @@
+  @completed "completed"
+  @failed "failed"
+
+  def new, do: @new
+  def processing, do: @processing
+  def completed, do: @completed
+  def failed, do: @failed
+
+  @default_format "csv"
+  @formats ["csv"]
+
+  @primary_key {:uuid, UUID, autogenerate: true}
+
+  schema "export" do
+    external_id(prefix: "exp_")
+
+    field(:schema, :string)
+    field(:format, :string, default: @default_format)
+    field(:status, :string)
+    field(:completion, :float)
+    field(:url, :string)
+    field(:filename, :string)
+    field(:path, :string)
+    field(:pid, :string)
+    field(:failure_reason, :string)
+    field(:estimated_size, :float)
+    field(:total_count, :integer)
+    field(:adapter, :string)
+    field(:params, :map)
+
+    belongs_to(
+      :user,
+      User,
+      foreign_key: :user_uuid,
+      references: :uuid,
+      type: UUID
+    )
+
+    belongs_to(
+      :key,
+      Key,
+      foreign_key: :key_uuid,
+      references: :uuid,
+      type: UUID
+    )
+
+    timestamps()
+    activity_logging()
+  end
+
+  defp create_changeset(changeset, attrs) do
+    changeset
+    |> cast_and_validate_required_for_activity_log(
+      attrs,
+      cast: [
+        :schema,
+        :status,
+        :completion,
+        :params,
+        :user_uuid,
+        :key_uuid
+      ],
+      required: [
+        :schema,
+        :status,
+        :completion,
+        :params
+      ]
+    )
+    |> validate_required_exclusive([:user_uuid, :key_uuid])
+    |> validate_inclusion(:format, @formats)
+    |> validate_inclusion(:status, [@new, @processing, @completed, @failed])
+    |> assoc_constraint(:user)
+    |> assoc_constraint(:key)
+  end
+
+  defp update_changeset(changeset, attrs) do
+    changeset
+    |> cast_and_validate_required_for_activity_log(
+      attrs,
+      cast: [
+        :status,
+        :completion,
+        :url,
+        :path,
+        :filename,
+        :adapter,
+        :schema,
+        :total_count,
+        :estimated_size,
+        :pid
+      ],
+      required: [
+        :status,
+        :completion
+      ]
+    )
+    |> validate_inclusion(:status, [@new, @processing, @completed, @failed])
+  end
+
+  def all_for(%User{} = user) do
+    from(t in Export, where: t.user_uuid == ^user.uuid)
+  end
+
+  def all_for(%Key{} = key) do
+    from(t in Export, where: t.key_uuid == ^key.uuid)
+  end
+
+  @doc """
+  Retrieves an export with the given ID.
+  """
+  @spec get(String.t(), keyword()) :: %Export{} | nil | no_return()
+  def get(id, opts \\ [])
+
+  def get(id, opts) when is_external_id(id) do
+    get_by([id: id], opts)
+  end
+
+  def get(_id, _opts), do: nil
+
+  @doc """
+  Retrieves an export using one or more fields.
+  """
+  @spec get_by(map() | keyword(), keyword()) :: %Export{} | nil | no_return()
+  def get_by(fields, opts \\ []) do
+    Export
+    |> Repo.get_by(fields)
+    |> preload_option(opts)
+  end
+
+  def init(export, schema, count, estimated_size, originator) do
```

Comment:
Can you add a function doc for this one? I'm not sure when to use `Export.init/5` vs. `Export.insert/1`.